docker/jenkins1-centos/contrib/kube-slave-common.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/sh
+#
+# This file provides functions to automatically discover suitable image streams
+# that the Kubernetes plugin will use to create "slave" pods.
+# The image streams has to have label "role" set to "jenkins-slave".
+#
+# The Jenkins container also need permissions to access the OpenShift API to
+# list image streams. You have to run this command to allow that:
+#
+# $ oc policy add-role-to-user edit system:serviceaccount:ci:default -n ci
+#
+# (where the 'ci' is the namespace where Jenkins runs)
+
+export DEFAULT_SLAVE_DIRECTORY=/tmp
+export SLAVE_LABEL="jenkins-slave"
+JNLP_SERVICE_NAME=${JNLP_SERVICE_NAME:-JENKINS_JNLP}
+JNLP_SERVICE_NAME=`echo ${JNLP_SERVICE_NAME} | tr '[a-z]' '[A-Z]' | tr '-' '_'`
+T_HOST=${JNLP_SERVICE_NAME}_SERVICE_HOST
+# the '!' handles env variable indirection so we can resolve the nested variable
+# see: http://stackoverflow.com/a/14204692
+JNLP_HOST=${!T_HOST}
+T_PORT=${JNLP_SERVICE_NAME}_SERVICE_PORT
+JNLP_PORT=${!T_PORT}
+
+export JNLP_PORT=${JNLP_PORT:-50000}
+
+ANDROID_SLAVE=docker.io/aerogear/jenkins-android-slave
+MAVEN_SLAVE=registry.access.redhat.com/aerogear/jenkins-android-slave
+# if the master is running the centos image, use the centos slave images.
+if [[ `grep CentOS /etc/redhat-release` ]]; then
+  ANDROID_SLAVE=docker.io/aerogear/jenkins-android-slave
+  MAVEN_SLAVE=openshift/jenkins-slave-maven-centos7
+fi
```

Comment:
@wtrocki - You can ignore my previous comment - this addresses the question :+1: